File: java/ch09/src/test/java/com/thesoftwaregorilla/tdd/money/FrancTest.java
package com.thesoftwaregorilla.tdd.money;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class FrancTest {

    @Test
    public void testMultiplication() {
        Money five = Money.franc(5);
        assertEquals(Money.franc(10), five.times(2));
        assertEquals(Money.franc(15), five.times(3));
    }

}

File: cs/ch05/Money.Tests/FrancTest.cs
namespace TheSoftwareGorilla.TDD.Money.Tests;

public class FrancTests
{
    
    [SetUp]
    public void Setup()
    {
        // No setup required for these tests
    }

    [Test]
    public void TestConstruction()
    {
        Franc five = new Franc(5);
        Assert.That(five, Is.Not.Null);
        Assert.That(five, Is.InstanceOf<Franc>());
        Assert.That(five, Is.EqualTo(new Franc(5)));
    }

    [Test]
    public void TestMultiplication()
    {
        Franc five = new Franc(5);
        Assert.That(new Franc(10), Is.EqualTo(five.Times(2)));
        Assert.That(new Franc(15), Is.EqualTo(five.Times(3)));
    }

    [Test]
    public void TestEquality()
    {
        Franc five = new Franc(5);
        Assert.That(new Franc(5), Is.EqualTo(five));
        Assert.That(new Franc(6), Is.Not.EqualTo(five));
    }
}